File: frameworks/domain.core/src/main/java/com/example/frameworks/domain/core/Value.java
package com.example.frameworks.domain.core;

public class Value<T> {

    private T value;

    public Value(T value, FieldType<T> fieldType) {
        this.value = new Payload(value).get(fieldType);
    }

    protected T get() {
        return value;
    }

    protected void set(T value) {
        this.value = value;
    }

}

File: frameworks/domain.core/src/main/kotlin/com/example/frameworks/domain/core/Value.kt
package com.example.frameworks.domain.core

open class Value<T : Any>(value: T, fieldType: FieldType<T>) {
    private var value: T

    init {
        this.value = Payload(value).get(fieldType)
    }

    protected fun get(): T {
        return value
    }

    protected fun set(value: T) {
        this.value = value
    }
}
